src/spacechop.ts

```diff
@@ -93,52 +98,67 @@ export const requestHandler = (
         return;
       }
 
+      res.set('X-Cache', 'HIT');
+      res.set('X-Key', fromCache.key);
       await respond(res, fromCache.stream, fromCache.contentType, config);
       trace.end();
       return;
     }
   }
 
   // look through sources to fetch original source stream
-  const stream = await lookThroughSources(sources, params);
+  const fromSource = await lookThroughSources(sources, params);
 
-  if (!stream) {
+  // It does not exist in source
+  if (fromSource && !fromSource.stream) {
     res.status(404);
     res.end('Could not find image');
     trace.warn('image', 'Could not find image');
     return;
+  } else {
+    res.set('X-Key', fromSource.key);
```

Comment:
Dont really think the 'X-Key' parameter is necessary, neither for cache hit or cache miss. What do you think is the use case?

Reply:
I use it to discover problematic images and to get an overview of how much re-use of original images a space has.

Reply:
What do you mean with problematic images? If i understand it correctly, for images from source, it only hashes the location of the image in the source. 

And for the re-use isn't that information carried in the 'X-Cache' header, whether it was a hit or not?

Reply:
Your thinking on the output end of things, I want to discover how much reusage there is on the input (source). Basically get a fact on how often within some time does an image get re-fetched from source, and by knowing that one can tune caching for sources to the correct size and time to live.

Reply:
Something I’ve not implemented is timing on the source download, this could be interesting to know about before adding cache or similar strategies because it might just be faster overall to implement those strategies, but I think it’s important to know this before putting an effort into adding any strategy.